### PR TITLE
Focus thunder skill

### PR DESCRIPTION
**Pull request by Pedrokase**

### CODE REVIEW

File: src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    private static boolean targetIsLivingEntity(Entity entity) {
+        return entity instanceof LivingEntity;
+    }
+
+    private static boolean targetNotArmorStand(Entity entity) {
+        return !(entity instanceof ArmorStand);
+    }
+
+    private static boolean targetHoldingSword(Entity entity) {
```
Comment (Phobossus) [inline src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java]: Doesn't it make sense to include all kind of metal tools, so not just swords but also pickaxes, shovels, whores?

File: src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                .addTrigger(TriggerType.VOCAL, new Trigger("bolta"))
+                .castRequirements(CAST_REQUIREMENTS)
+                .build();
+    }
+
+    @Override
+    protected void invoke(WorldEntity caster) {
+        World world = caster.getLocation().getWorld();
+
+        Collection<Entity> result = world.getNearbyEntities(
```
Comment (Phobossus) [inline src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java]: Can be collected straight to a list if its needed.

File: src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                caster.getLocation(),
+                BLOCK_RADIUS,
+                BLOCK_RADIUS,
+                BLOCK_RADIUS,
+                nearbyEntitiesFilter
+        );
+
+        ArrayList<Entity> nearbyEntities = new ArrayList<>(result);
+        Random random = new Random();
+        for (int i = 0; i < MAX_TARGETS; i++) {
```
Comment (Phobossus) [inline src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java]: Move to a "strikeRandomTargets" method.

File: src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            if (nearbyEntities.isEmpty()) return;
```
Comment (Phobossus) [inline src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java]: Why do we check for nearbyEntities existence in each iteration? If there are no nearby entities, then there doesn't seem to be a reason to execute the for() at all.

File: src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java (hunk 1 above)
Comment (Pedrokase) [reply src/main/java/abilitiesV2/skills/abitilies/manipulation/FocusThunder.java]: Yea, I can do that

**Pull request closed without merging**